new-lamassu-admin/src/pages/Blacklist/Blacklist.js

```diff
@@ -0,0 +1,141 @@
+import { useQuery, useMutation } from '@apollo/react-hooks'
+import Grid from '@material-ui/core/Grid'
+import { makeStyles } from '@material-ui/core/styles'
+import gql from 'graphql-tag'
+import * as R from 'ramda'
+import React, { useState } from 'react'
+
+import { Link } from 'src/components/buttons'
+import Sidebar from 'src/components/layout/Sidebar'
+import TitleSection from 'src/components/layout/TitleSection'
+
+import styles from './Blacklist.styles'
+import BlackListModal from './BlacklistModal'
+import BlacklistTable from './BlacklistTable'
+
+const useStyles = makeStyles(styles)
+
+const DELETE_ROW = gql`
+  mutation DeleteBlacklistRow($cryptoCode: String, $address: String) {
+    deleteBlacklistRow(cryptoCode: $cryptoCode, address: $address)
+  }
+`
+
+const GET_BLACKLIST = gql`
+  query getBlacklistData {
+    blacklist {
+      cryptoCode
+      address
+    }
+    cryptoCurrencies {
+      display
+      code
+    }
+  }
+`
+
+const SAVE_CONFIG = gql`
+  mutation Save($config: JSONObject) {
+    saveConfig(config: $config)
+  }
+`
+
+const GET_INFO = gql`
+  query getData {
+    config
+  }
+`
+
+const ADD_ROW = gql`
+  mutation InsertBlacklistRow($cryptoCode: String!, $address: String!) {
+    insertBlacklistRow(cryptoCode: $cryptoCode, address: $address) {
+      cryptoCode
+      address
+    }
+  }
+`
+
+const Blacklist = () => {
+  const { data: blacklistResponse } = useQuery(GET_BLACKLIST)
+  const { data: configData } = useQuery(GET_INFO)
+  const blacklistData = R.path(['blacklist'])(blacklistResponse) ?? []
+  const availableCurrencies =
+    R.path(['cryptoCurrencies'], blacklistResponse) ?? []
+
```

Comment:
Hooks have a bunch of rules for them to work properly, because of this I tend to prefer to get group them on the top of the function. 

Try to move all hooks to the top and the logic related to them afterwards.